docs/how_to_use/how_to_migrate_your_circuit.md

```diff
@@ -0,0 +1,132 @@
+
+# How to migrate you circuit in Rust
+
+Halo2 is currently the most widely used Proving System, with implementations such as [zcash/halo2](https://github.com/zcash/halo2), [privacy-scaling-explorations/halo2](https://github.com/privacy-scaling-explorations/halo2), and [scroll-tech/halo2](https://github.com/scroll-tech/halo2).
+Proving programs equipped with a Halo2 can experience a decrease in proof generation costs when switching the backend to Tachyon.
+
+## Requirements
+
+1. Replace `halo2` library in your program with [tachyon-halo2](https://github.com/kroma-network/halo2).
+2. Install the tachyon debian package, see *Debian packaging* section in [how_to_build.md](./how_to_build.md)
+
+## Your Circuit
+
+Let's assume that your circuit has an interface almost identical to the [example circuit](../../vendors/halo2/src/circuits/shuffle_circuit.rs) we have prepared in advance.
+
+``` rust
+let your_circuit = YourCircuit<_>;
+let your_instances = <YourInstances>
+```
+
+What this document ultimately aims to describe is how to call the `create_tachyon_proof()` function.
+
+## ProvingKey for Tachyon
+
+If your prover program is utilizing the halo2 library, it would be have generated the proving key as follows. (In a production level, the `params` would be initialized by reading from a file)
+
+``` rust
+use halo2_proofs::plonk::halo2_proofs::{
+    plonk::{keygen_pk, keygen_pk2},
+    poly::ParamsKZG,
+use halo2curves::bn256::Fr;
+
+// |k|: log of degree, |s|: toxic-waste secret
+let k: u32 = 5;
+let s = Fr::from(2);
+
+// Generate |params|
+let params = ParamsKZG<Bn256>::unsafe_setup_with_s(k, s);
+let params: Option<ParamsKZG<Bn256>> = Some(params);
+
+// Generate |pk| as ProvingKey
+let pk = keygen_pk2(params.as_ref().unwrap(), &circuit).unwrap();
+```
+
+A `tachyon_pk` that can be recognized by the tachyon library can be created as follows. In the final step, supply `tachyon_pk` as an argument when calling `create_tachyon_proof()`.
+
+``` rust
+use halo2_proofs::bn254::ProvingKey as TachyonProvingKey;
+
+let mut tachyon_pk = {
+    let mut pk_bytes: Vec<u8> = vec![];
```

Comment:
```suggestion
    // Please remember to release pk_bytes immediately after deserialization, as it can occupy tens of gigabytes of memory.
    let mut pk_bytes: Vec<u8> = vec![];
```